gammapy/data/gti.py

```diff
@@ -116,4 +116,20 @@ def select_time(self, time_interval):
         gti : `GTI`
             Copy of the GTI table with selection applied.
         """
-        raise NotImplementedError
+        # get GTIs that fall within the time_interval
+        mask = self.time_start >= time_interval[1]
+        mask |= self.time_stop <= time_interval[0]
+        gti_within = self.table[~mask]
+
+        # crop the GTIs
```

Comment:
For cropping I find `np.clip()`a bit easier to read...

Reply:
yes, thanks, it reads much better now!